### PR TITLE
Add in basic but functional search history

### PR DESCRIPTION
**Pull request by Tehsurfer**
Add search history

Note that you need 6 unique search history items to get the select box to appear
image

### CODE REVIEW

File: src/components/SearchHistory.vue [hunk 1]
```diff
@@ -0,0 +1,145 @@
+<template>
+  <div class="container">
+    <!-- <span v-if="reversedSearchHistory.length > 0" class="title"> Search History </span> -->
+    <template v-for="(item, i) in reversedSearchHistory">
+      <el-tag class="search-tag" v-if="i < 3" v-bind:key="i" @click="search(item)">{{ item.search }} </el-tag>
+    </template>
+    <el-select  
+      v-if="reversedSearchHistory.length > 6"
```
[the remark this answers is not in the document]
Comment (alan-wu) [reply src/components/SearchHistory.vue:8]: It is also confusing to have the select showing a search term that is not currently active when I am doing a new search using the search input.

File: src/components/SearchHistory.vue [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      v-if="reversedSearchHistory.length > 6" 
+      :value="selectValue" 
+      class="m-2 search-select" 
+      placeholder="Full search History" 
+      size="small"
+      popper-class="sidebar-search-select-popper"
+      @change="selectChange"
+    >
+      <el-option
+        v-for="(item, i) in cascaderOptions"
```
[the remark this answers is not in the document]
Comment (alan-wu) [reply src/components/SearchHistory.vue:17]: I think a customised autocomplete in the search input may work better.

**Review by alan-wu — CHANGES_REQUESTED** [review]
Is it possible to add a popup showing the facets for each of the tag/search result?
I can have multiple heart results with different facets but I cannot tell them apart.

File: src/components/SearchHistory.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (Tehsurfer) [reply src/components/SearchHistory.vue:17]: I have removed the filters.

For autocomplete, i will create another version and send it to dominic (he asked for this design)

File: src/components/SearchHistory.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tehsurfer) [reply src/components/SearchHistory.vue:8]: Will reset the placeholder and always show the select box

File: src/components/SearchHistory.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (Tehsurfer) [reply src/components/SearchHistory.vue:17]: @alan-wu autocomplete is here:

https://github.com/ABI-Software/map-sidebar/pull/65

[the remark this answers is not in the document]
Comment (Tehsurfer) [reply src/components/SearchHistory.vue:17]: I asked Dominic if that is better

**Pull request merged**